src/Router.sol

```diff
@@ -312,7 +312,7 @@ contract Router is IRouter, Ownable, EIP712 {
             if (block.timestamp > deadline) revert SignatureExpired(deadline);
             if (!user.isValidSignatureNow(_hashTypedDataV4(details._hash()), userSignature)) revert InvalidSignature();
             if (executionNonces[user] != nonce) revert InvalidNonce();
-            executionNonces[user] = nonce + 1;
+            ++executionNonces[user];
```

Comment:
one more nonce in `src/library/Delegation.sol`. please help fix it too. Thanks.

Reply:
Nonce does not guarantee to be the same in that function. I don't think it's proper to do that.